### ISSUE TITLE
Incompatibility with cryptography 42.0.0

### ISSUE THREAD

**Issue by DBS-ST-VIT** [scheiblingco/sshkey-tools#61]
Hi,
recently, there was a new major release of the cryptography package, which is a dependency of this project. Using the new major version combined with `sshkey_tools.keys.Ed25519PrivateKey`, i am receiving the following error:

```
  File "/root/.local/share/pipx/venvs/config-updater/lib64/python3.9/site-packages/config_updater/ssh_keygen.py", line 2, in <module>
    from sshkey_tools.keys import (
  File "/root/.local/share/pipx/venvs/config-updater/lib64/python3.9/site-packages/sshkey_tools/keys.py", line 10, in <module>
    from cryptography.hazmat.backends.openssl.ec import (
ModuleNotFoundError: No module named 'cryptography.hazmat.backends.openssl.ec'
```

I guess its may be a quite risky idea, to allow new major versions without any pre-tests, like given here: https://github.com/scheiblingco/sshkey-tools/blob/main/requirements.txt#L2C1-L2C19

**Comment by scheibling** [comment]
@DBS-ST-VIT I vividly remember myself doing this when I encountered this last time (<>41), but maybe that was in a parallell universe somewhere...

Jokes aside, you're absolutely right, I'll have a look later this afternoon. Thanks for the notice!

**Comment by DBS-ST-VIT** [comment]
Yeah we just saw it this morning in context of an production outage with some dozens of users affected, but that was a result of a fuckup with `poetry build` and pinning of implicit dependencies on my side, not directly the fault of you/this project. :P

Thanks for the quick response and your support! :)

**Comment by scheibling** [comment]
Currently uploading the fix, will be the next subminor 
I'll add the 42.x compability later on as well, probably in the next minor at the same time as the data signature functionality

I'm just happy this is being used, especially since the project I originally wrote it for originally didn't end up being written (not yet anyway)

**Comment by scheibling** [comment]
@DBS-ST-VIT Seems it was just a case of updating to the newer rust bindings (probably, need to test that a bit more to be sure nothing else has changed before release). If you wanna test it or have a look, it's in the crpytography-v42 branch

**Comment by barrybarrette** [comment]
> @DBS-ST-VIT Seems it was just a case of updating to the newer rust bindings (probably, need to test that a bit more to be sure nothing else has changed before release). If you wanna test it or have a look, it's in the crpytography-v42 branch

I've got a production dependency on this project as well, so thanks for all your great work on this. I can confirm this branch works for my simple use case of generating Ed25519 keys and exporting them as strings. Happy to help with whatever testing needs to be done to get this branch released.

Thanks again!

**Comment by mbarr-hrp** [comment]
Any idea when this will be released?

Looks like this was also picked up in https://github.com/scheiblingco/sshkey-tools/pull/74

**Comment by scheibling** [comment]
@mbarr-hrp Sorry, I forgot about this completely. Give me a couple of hours

**Comment by scheibling** [comment]
Going through all of the testing for release now, when it's done version 0.11.1 will be available for use with cryptography 42.0.x

Double-checked this time that the version is actually properly pinned, so the issue should, hopefully, not reoccur :-)
